torch/utils/benchmark/utils/compile.py

```diff
@@ -34,36 +35,37 @@ def _disable_tensor_cores():
     def bench_loop(
         model: Union[torch.nn.Module, Callable],
         sample_input: Union[torch.Tensor, Any],
-        num_iters: int,
+        num_iters: int = 5,
         optimizer: torch.optim.Optimizer = None,
         loss_fn: Callable = None,
     ):
-        """
-        This is a simple loop that can be used to benchmark a model for either training or inference
-        It takes care of taking several measurements and averaging them
-        It also takes care of calling `cuda.synchronize()` if the model is on GPU
-        """
-        durations = []
-        for _ in range(num_iters):
-            start = time.time()
-
-            if optimizer:
-                optimizer.zero_grad()
-                output = model(sample_input)
-                loss = loss_fn(output) if loss_fn else output.sum()
-                loss.backward()
-                optimizer.step()
-            else:
-                model(sample_input)
+        # Define the statement and setup for the benchmark
+        if optimizer and loss_fn:
+            # Training mode
+            stmt = """
+    output = model(sample_input)
+    loss = loss_fn(output) if loss_fn else output.sum()
+    loss.backward()
+    optimizer.step()
+    optimizer.zero_grad()
+            """
+        else:
+            # Inference mode
+            stmt = "model(sample_input)"
+
+        # Create the Timer object
+        timer = Timer(
+            stmt=stmt,
+            globals={"model": model, "sample_input": sample_input, "optimizer": optimizer, "loss_fn": loss_fn}
```

Comment:
loss_fn and optimizers are not necessarily assigned?

Reply:
I might be missing something but they both default to None values, would you prefer if I explicitly checked if they were None before constructing the globals dict?

Reply:
You are right, that's fine